test/e2e/samples.go

```diff
@@ -110,48 +112,105 @@ func retrieveBuildAndBuildRun(namespace string, buildRunName string) (*operator.
 }
 
-// outputBuildAndBuildRunStatusAndPodLogs will output the status of build and buildRun, print logs of taskRun pod
-func outputBuildAndBuildRunStatusAndPodLogs(namespace string, buildRunName string) {
+// printTestFailureDebugInfo will output the status of Build, BuildRun, TaskRun and Pod, also print logs of Pod
+func printTestFailureDebugInfo(namespace string, buildRunName string) {
+	Logf("Print failed BuildRun's log")
+
 	f := framework.Global
-	buildRun, build, err := retrieveBuildAndBuildRun(namespace, buildRunName)
-	Expect(err).ToNot(HaveOccurred(), "Failed to retrieve build and buildRun")
-	Logf("The status of Build %s: %s, %s", build.Name, build.Status.Reason, build.Status.Registered)
-	Logf("The status of BuildRun %s: %s", buildRun.Name, buildRun.Status.Succeeded)
-	Logf("The reason of BuildRun %s: %s", buildRun.Name, buildRun.Status.Reason)
 
-	lbls := map[string]string{
-		operator.LabelBuildRun: buildRunName,
-	}
-	listOptions := client.ListOptions{
-		Namespace:     namespace,
-		LabelSelector: labels.SelectorFromSet(lbls),
+	buildRun, build, err := retrieveBuildAndBuildRun(namespace, buildRunName)
+	if build != nil {
+		Logf("The status of Build %s: registered=%s, reason=%s", build.Name, build.Status.Registered, build.Status.Reason)
+		if buildJson, err := json.Marshal(build); err == nil {
+			Logf("The full Build: %s", string(buildJson))
+		}
 	}
 
-	podList := &v1.PodList{}
-	err = f.Client.List(context.TODO(), podList, &listOptions)
-	Expect(err).ToNot(HaveOccurred(), "Failed to retrieve pods.")
-	Expect(len(podList.Items)).To(Equal(1), "Did not retrieve one pod.")
-
-	pod := &podList.Items[0]
-
-	for _, container := range pod.Spec.Containers {
-		req := f.KubeClient.CoreV1().Pods(namespace).GetLogs(pod.Name, &v1.PodLogOptions{
-			TypeMeta:  metav1.TypeMeta{},
-			Container: container.Name,
-			Follow:    false,
-		})
-		podLogs, err := req.Stream()
-		Expect(err).ToNot(HaveOccurred(), "Failed to retrieve logs of container.")
-		buf := new(bytes.Buffer)
-		_, err = io.Copy(buf, podLogs)
-		Expect(err).ToNot(HaveOccurred(), "Failed to copy container logs to buffer.")
-		strLogs := buf.String()
-		Logf("Logs of container %s: %s", container.Name, strLogs)
+	if buildRun != nil {
+		Logf("The status of BuildRun %s: succeeded=%s, reason=%s", buildRun.Name, buildRun.Status.Succeeded, buildRun.Status.Reason)
+		if buildRunJson, err := json.Marshal(buildRun); err == nil {
+			Logf("The full BuildRun: %s", string(buildRunJson))
+		}
+
+		podName := ""
+
+		// Only log details of TaskRun if Tekton objects can be accessed
+		if os.Getenv(EnvVarVerifyTektonObjects) == "true" {
+			if taskRun, _ := getTaskRun(f, buildRun); taskRun != nil {
+				reason := ""
+				message := ""
+				condition := taskRun.Status.GetCondition(apis.ConditionSucceeded)
+				if condition != nil {
+					reason = condition.Reason
+					message = condition.Message
+				}
+
+				Logf("The status of TaskRun %s: reason=%s, message=%s", taskRun.Name, reason, message)
```

Comment:
I would have put this one inside:

```go
				if condition != nil {
					Logf("The status of TaskRun %s: reason=%s, message=%s", taskRun.Name, condition.Reason, condition.Message)
			
				}
```
in order to avoid creating the `reason` and `message` vars.

Reply:
Makes sense. :-)

Reply:
Done.